### DISCUSSION TITLE
Issue after migrating Angular CLI to NX: NgRx/Signals store error

### DISCUSSION THREAD

**Discussion by t1m4lc** [ngrx/platform#4525] [Q&A]
Hi everyone,

I recently migrated my project from Angular CLI to NX, and I've started encountering some strange errors with the NgRx/Signals store that I didn't have before. Specifically, I'm seeing the following error:

```
Argument of type 'unknown' is not assignable to parameter of type 'WritableStateSource<object>'.ts(2345)
or 
Exported variable 'UserStore' has or is using name 'SIGNAL' from external module "project/node_modules/@angular/core/index" but cannot be named.ts(4023)
⚠ Error (TS4023) Exported variable UserStore has or is using name SIGNAL from external module 
```
![Screenshot from 2024-09-17 19-10-44](https://github.com/user-attachments/assets/e1f50f52-2dad-47de-8764-4060894a12c4)

I've attached screenshots of the error and relevant code sections. Has anyone experienced similar issues or know what might be causing this after the migration? Any guidance on how to resolve it would be greatly appreciated!

Thanks in advance for your help!

PS: v18.2.0 for ANgular and @ngrx/signals

**Comment by brandonroberts** [comment]
https://github.com/ngrx/platform/issues/4151

**Comment by rainerhahnekamp** [comment]
@alcaidio are you accessing the SignalStore from an Nx Lib, right? Can you show a screenshot from the place you export it and where you use it?

**Reply by t1m4lc** [reply]
Yes it's on Nx, but its' just side effect errors... Thanks @rainerhahnekamp

**Comment by t1m4lc** [comment]
Edit: Maybe it was a side effect linked to lint/build errors that had not been resolved in a number of files. Once resolved, the second error disappeared, but not the first one when i want to use the store in patchState in an nx lib (not buildable)

**Reply by t1m4lc** [reply]
The error occur in all my nx libs. 

```
Argument of type '{ ...}>'.
  Types of property '[STATE_SOURCE]' are incompatible.
    Type 'Signal<{ ... }>': set, update, asReadonly, [ɵWRITABLE_SIGNAL]
```

**Reply by rainerhahnekamp** [reply]
Can you provide a snippet?

**Reply by t1m4lc** [reply]
How can i provide you a snippet ?

**Reply by rainerhahnekamp** [reply]
That error message must come from some part of your code that you've written. Can you provide that or is it failing compile the SignalStore library?

**Reply by t1m4lc** [reply]
Sorry for the late reply, here are two screenshots, I hope that's enough? 
![Screenshot from 2024-10-02 11-03-57](https://github.com/user-attachments/assets/dd8ea14f-7e70-46d1-98b7-7d662f670153)
![Screenshot from 2024-10-02 11-04-21](https://github.com/user-attachments/assets/b2450628-9847-4504-b139-b99c28492483)

**Comment by rainerhahnekamp** [comment]
@brandonroberts will cae429ad6c903ea0be3984e8572d49f2a4ba248c fix that? Looks that way.
